tool/msh3_app.cpp

```diff
@@ -83,5 +106,9 @@ int MSH3_CALL main(int argc, char **argv) {
         MsH3ApiClose(Api);
     }
 
+    if(givenHost != NULL){
```

Comment:
You don't need to check for NULL, since `free` does itself.

Reply:
Fixed at ddd6b31